### ISSUE TITLE
Build failure: tensorflow-datasets

### ISSUE THREAD

**Issue by dmayle** [NixOS/nixpkgs#226947]
### Steps To Reproduce

1. build *tensorflow-datasets*

### Build log

https://gist.github.com/dmayle/9888851faa7860e465910a934ddef41a

### Additional context

Python packages `click` and `psutil` were added as requirements for tensorflow-datasets 4.7.0, but missed in the package bump by @mweinelt 

### Notify maintainers
@ndl 

### Metadata

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
output here
```

**Comment by dmayle** [comment]
I've noticed that I was trying to build python310Packages.tensorflow-datasets, but it seems that apache-beam isn't even supported in python310Packages, so I'm trying to rebuild with python39Packages and will update this bug in a couple of hours (after tensorflowWithCude finishes compiling)

**Comment by dmayle** [comment]
Here's a gist for python39Package.  Again, missing the `click` and `psutil` dependencies:
https://gist.github.com/dmayle/367d541711ed32f2f964cdfaa650962b

**Comment by dmayle** [comment]
BTW - I have an overlay that overrides the build to make it work:
```
{ pkgs ? import <nixpkgs> {} }:
let
  python = let
    packageOverrides = self: super: {
      tensorflow-datasets = super.tensorflow-datasets.overridePythonAttrs(old: rec {
        propagatedBuildInputs = old.propagatedBuildInputs ++ (with super; [
          absl-py
          click
          etils
          psutil
          toml
          wrapt
        ]);
        nativeCheckInputs = old.nativeCheckInputs ++ (with super; [
          datasets
          pyyaml
        ]);
        disabledTestPaths = builtins.filter (p:
        p != "tensorflow_datasets/audio/groove_test.py" && p != "tensorflow_datasets/audio/nsynth_test.py")
        old.disabledTestPaths ++ [
          "tensorflow_datasets/import_without_tf_test.py"
          "tensorflow_datasets/core/registered_test.py"
          "tensorflow_datasets/core/dataset_utils_test.py"
          "tensorflow_datasets/core/dataset_builders/conll/conllu_dataset_builder_test.py"
          "tensorflow_datasets/core/dataset_builders/huggingface_dataset_builder_test.py"
          "tensorflow_datasets/core/features/audio_feature_test.py"
          "tensorflow_datasets/core/features/sequence_feature_test.py"
          "tensorflow_datasets/datasets/groove/groove_dataset_builder_test.py"
          "tensorflow_datasets/datasets/universal_dependencies/universal_dependencies_dataset_builder_test.py"
          "tensorflow_datasets/datasets/xtreme_pos/xtreme_pos_dataset_builder_test.py"
          "tensorflow_datasets/datasets/nsynth/nsynth_dataset_builder_test.py"
        ];
      });
    };
  in pkgs.python39.override {inherit packageOverrides; self = python;};
in (python.withPackages (p: with p; [ tensorflow-datasets ])).env;
```

**Comment by natsukium** [comment]
closed by #227127
